### PR TITLE
Throw error for unsupported file types

### PR DESCRIPTION
**Pull request by djalova**
Fix for #3

### CODE REVIEW

File: api/model.py [hunk 1]
```diff
@@ -50,9 +51,13 @@ def post(self):
         result = {'status': 'error'}
 
         args = image_parser.parse_args()
+
+        if not args['image'].mimetype.endswith(('jpg', 'jpeg', 'png')):
+            abort(400, 'Invalid file type/extension')
```
Comment (MLnick) [inline api/model.py]: Perhaps add that only JPEG and PNG are supported in the error message? Such as "Invalid file type/extension. Please provide a JPEG or PNG image" or similar.

**Comment by djalova** [comment]
@MLnick I updated the message

**Comment by MLnick** [comment]
Happy to merge but maybe @ptitzler wants to take a quick look

**Comment by MLnick** [comment]
The input validation is handled by the request parser: https://github.com/IBM/MAX-Image-Caption-Generator/blob/master/api/model.py#L38.

Since it is a required param the following is returned:

```
400 
{"errors": {"image": "Missing required parameter in an uploaded file"}, "message": "Input payload validation failed"}
```

**Comment by MLnick** [comment]
Though I must say that Flask-restplus built in error could perhaps be improved! ;)

**Comment by djalova** [comment]
I'll take a look at updating that error.

**Comment by MLnick** [comment]
it’s built into flask-restplus itself - but we could open a PR there if we
feel strongly

**Comment by ptitzler** [comment]
Meh, not feeling strongly enough about it because we don't own that code. As long as we return the proper status code whenever any kind of invalid input is provided  I think you can go ahead and merge.

**Comment by MLnick** [comment]
@ajbozarth hmm, yeah I guess we've always been using cURL to test (at least I have) and so mimetype was set. But often it may not be.

I wonder if perhaps the `mimetype` fix should also be rolled into the API here (i.e. if mimetype is not set, attempt to get it from the file and fail only if that also doesn't work).

@djalova @ptitzler ?

**Comment by ptitzler** [comment]
Wouldn't hurt for sure to do that!

**Comment by ajbozarth** [comment]
I opened issue #13 to address mime type in the model, we can either address it or choose not to and close as "will not fix"

**Pull request merged**